digit-recognition/frontend/src/features/digit-recognizer/utils.ts

```diff
@@ -19,18 +19,26 @@ export const getFlattenedPixelArray = (
 
   const imageData = tempCtx.getImageData(0, 0, expectedSize, expectedSize);
 
-  const pixels = imageData.data; // Array of RGBA values ​​(4 elements per pixel)
+  const pixels = imageData.data; // Array of RGBA values (4 elements per pixel)
   const grayscaleArray = [];
 
   for (let i = 0; i < pixels.length; i += 4) {
-    const r = pixels[i];
-    const g = pixels[i + 1];
-    const b = pixels[i + 2];
-    const brightness = Math.round((r + g + b) / 3);
-
-    grayscaleArray.push(brightness > 0 ? 255 : 0);
+    if (pixels[i] === 0) {
+      grayscaleArray.push(0);
+      continue;
+    }
+
+    const getRandomDeviation = () => Math.round(Math.random() * 10 - 5);
+    const isUseDeviation = Math.random() > 0.5;
+    const middleBrightness = 250;
+    const noisedBrightness = isUseDeviation
+      ? middleBrightness + getRandomDeviation()
+      : middleBrightness;
+
+    grayscaleArray.push(Math.min(noisedBrightness, 255));
   }
 
+  console.log("grayscaleArray:", grayscaleArray.join(", "));
```

Comment:
Instead of a function to have log side effect, better to do it in the component where function is called.